flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogDatabase.java

```diff
@@ -29,259 +32,227 @@
     sql("USE CATALOG %s", catalogName);
-
-    Assert.assertTrue(
-        "Namespace should exist", validationNamespaceCatalog.namespaceExists(icebergNamespace));
+    assertThat(validationNamespaceCatalog.namespaceExists(icebergNamespace))
+        .as("Namespace should exist")
+        .isTrue();
 
     List<Row> databases = sql("SHOW DATABASES");
 
     if (isHadoopCatalog) {
-      Assert.assertEquals("Should have 1 database", 1, databases.size());
-      Assert.assertEquals("Should have db database", "db", databases.get(0).getField(0));
-
+      assertThat(databases).hasSize(1);
+      assertThat(databases.get(0).getField(0)).as("Should have db database").isEqualTo("db");
       if (!baseNamespace.isEmpty()) {
         // test namespace not belongs to this catalog
         validationNamespaceCatalog.createNamespace(
             Namespace.of(baseNamespace.level(0), "UNKNOWN_NAMESPACE"));
         databases = sql("SHOW DATABASES");
-        Assert.assertEquals("Should have 1 database", 1, databases.size());
-        Assert.assertEquals(
-            "Should have db and default database", "db", databases.get(0).getField(0));
+        assertThat(databases).hasSize(1);
+        assertThat(databases.get(0).getField(0)).as("Should have db database").isEqualTo("db");
       }
     } else {
       // If there are multiple classes extends FlinkTestBase, TestHiveMetastore may loose the
       // creation for default
       // database. See HiveMetaStore.HMSHandler.init.
-      Assert.assertTrue(
-          "Should have db database",
-          databases.stream().anyMatch(d -> Objects.equals(d.getField(0), "db")));
+      assertThat(databases)
+          .anyMatch(d -> Objects.equals(d.getField(0), "db"))
+          .as("Should have db database");
     }
   }
 
-  @Test
+  @TestTemplate
   public void testCreateNamespaceWithMetadata() {
-    Assume.assumeFalse("HadoopCatalog does not support namespace metadata", isHadoopCatalog);
-
-    Assert.assertFalse(
-        "Namespace should not already exist",
-        validationNamespaceCatalog.namespaceExists(icebergNamespace));
-
+    assumeThat(isHadoopCatalog).as("HadoopCatalog does not support namespace metadata").isFalse();
+    assertThat(validationNamespaceCatalog.namespaceExists(icebergNamespace))
+        .as("Namespace should not already exist")
+        .isFalse();
     sql("CREATE DATABASE %s WITH ('prop'='value')", flinkDatabase);
-
-    Assert.assertTrue(
-        "Namespace should exist", validationNamespaceCatalog.namespaceExists(icebergNamespace));
-
+    assertThat(validationNamespaceCatalog.namespaceExists(icebergNamespace))
+        .as("Namespace should exist")
+        .isTrue();
     Map<String, String> nsMetadata =
         validationNamespaceCatalog.loadNamespaceMetadata(icebergNamespace);
-
-    Assert.assertEquals(
-        "Namespace should have expected prop value", "value", nsMetadata.get("prop"));
+    assertThat(nsMetadata.get("prop"))
```

Comment:
better to use `assertThat(nsMetadata).containsEntry("prop", "value")` as that will print the content of the map if the assertion ever fails